src/LocationSelector/index.js

```diff
@@ -1,39 +1,43 @@
 import React from "react";
 import { geocodeByLatlng, getAddressComponent, isFunction } from "../utils";
-import MapWithSearch from "../MapWithSearch";
+import LocationSelectorMapWithSearch from "../MapsWithSearch/LocationSelectorMap";
 
 class LocationSelector extends React.Component {
+  geocodeAndInformParent = async latLng => {
+    const results = await geocodeByLatlng(latLng);
+    if (results && results.length) {
+      const result = results[0];
+      if (isFunction(latLng.lat)) {
+        latLng = { lat: latLng.lat(), lng: latLng.lng() };
+      }
+      this.props.onSearchSuccess({
+        latLng,
+        name: "",
+        address: result.formatted_address,
+        place_id: result.place_id,
+        types: result.types,
+        locality: getAddressComponent(
+          result.address_components,
+          "neighborhood"
+        ),
+        city: getAddressComponent(result.address_components, "locality"),
+        country: getAddressComponent(result.address_components, "country")
```

Comment:
not super necessary, but we can store all these strings `neighborhood`, `locality` , `country` in a an enum. there's a constant.js file in `src`